lib/system/mmdisp.nim

```diff
@@ -155,9 +157,14 @@ when defined(boehmgc):
     when hasThreadSupport:
       boehmGC_allow_register_threads()
 
+  proc boehmgc_finalizer(obj: pointer, typedFinalizer: proc(x: pointer)) =
```

Comment:
Don't use a .closure proc here, unnecessary overhead.

Reply:
boehm gc's finalizer function has a different signature, an extra `void* clientdata`, do you mean omit the extra one?

Reply:
Then it shall be ``typedFinalizer: (proc(x, clientData: pointer) {.cdecl.})``

Reply:
Nim `typ.finalizer` signature is: `f(x: pointer)`, boehm need a finalizer with signature: `f(x, clientData: pointer)`, so seems need a proxy proc like the one: `boehmgc_finalizer`.

do you mean declare proxy proc as `{.cdecl.}`?  because `clientData: pointer` not belongs to  `typedFinalizer`.

Reply:
Or is fine like this?  `typedFinalizer: (proc(x: pointer) {.cdecl.})`